clang/tools/clang-offload-wrapper/ClangOffloadWrapper.cpp

```diff
@@ -685,19 +754,45 @@ class BinaryWrapper {
     auto *i32Zero = ConstantInt::get(Type::getInt32Ty(C), 0u);
     auto *NullPtr = Constant::getNullValue(Type::getInt8PtrTy(C));
 
-    Expected<MemoryBuffer *> MBOrErr = loadFile(EntriesFile);
-    if (!MBOrErr)
-      return MBOrErr.takeError();
-    MemoryBuffer *MB = *MBOrErr;
-
     std::vector<Constant *> EntriesInits;
     // Only the name field is used for SYCL now, others are for future OpenMP
     // compatibility and new SYCL features
-    for (line_iterator LI(*MB); !LI.is_at_eof(); ++LI)
-      EntriesInits.push_back(ConstantStruct::get(
-          getEntryTy(), NullPtr,
-          addStringToModule(*LI, "__sycl_offload_entry_name"), Zero, i32Zero,
-          i32Zero));
+    if (SymProp_Table) {
+      Constant *Entries_Name =
+          Current_DeviceImage_Initializer->getAggregateElement(
+              10); // EntriesBegin field of struct __tgt_device_image
```

Comment:
Where do these hardcoded values come from?

Reply:
Updated to use constexpr symbolic constants..